app/controllers/concerns/course/discussion/posts_concern.rb

```diff
@@ -30,6 +38,17 @@ def destroy
   # Render a new post in a separate page
   # This can be used when you do not want to introduce javascript to your implementation
   def reply
+    @reply_post = discussion_topic.posts.build(title: t('course.discussion.posts.reply_title',
+                                                        title: @post.title))
```

Comment:
`@reply_post` and `@post` are not the same thing. `@reply_post` is a new post which replies to an existing `@post`.